File: scripts/algorithms/Water and Jug Problem/Water and Jug Problem.java
class Solution {
    private static int gcd(int a,int b){
        if(b==0)return a;
        return gcd(b,a%b);
    }
    public boolean canMeasureWater(int jug1Capacity, int jug2Capacity, int targetCapacity) {
        if(targetCapacity>jug1Capacity+jug2Capacity){
            return false;
        }
        int g=gcd(jug1Capacity,jug2Capacity);
        return (targetCapacity%g==0);
    }
}

File: algorithms/W/Water and Jug Problem/Water and Jug Problem.js
var canMeasureWater = function(jug1Capacity, jug2Capacity, targetCapacity) {
	const gcd = (x, y) => y === 0 ? x : gcd(y, x % y);

	return jug1Capacity + jug2Capacity >= targetCapacity && 
		targetCapacity % gcd(jug1Capacity, jug2Capacity) === 0;
};